homeassistant/components/weather/__init__.py

```diff
@@ -332,38 +330,55 @@ def add_to_platform_start(
-                        "WeatherEntity, this is deprecated and will be unsupported "
-                        "from Home Assistant 2024.3. Please %s"
-                    ),
-                    self.__module__,
-                    self.entity_id,
-                    report_issue,
-                )
-                ir.async_create_issue(
-                    self.hass,
-                    DOMAIN,
-                    f"deprecated_weather_forecast_{self.platform.platform_name}",
-                    breaks_in_ha_version="2024.3.0",
-                    is_fixable=False,
-                    is_persistent=False,
-                    issue_domain=self.platform.platform_name,
-                    severity=ir.IssueSeverity.WARNING,
-                    translation_key="deprecated_weather_forecast",
-                    translation_placeholders={
-                        "platform": self.platform.platform_name,
-                        "report_issue": report_issue,
-                    },
-                )
-                _reported_forecast = True
+        if self.__weather_legacy_forecast:
+            self._report_legacy_forecast(hass)
+
+    def _report_legacy_forecast(self, hass: HomeAssistant) -> None:
+        """Log warning and create an issue if the entity imlpements legacy forecast."""
+        if "custom_components" not in type(self).__module__:
+            # Do not report core integrations as they are already fixed or PR is open.
+            return
+
+        report_issue = async_suggest_report_issue(
+            hass,
+            integration_domain=self.platform.platform_name,
+            module=type(self).__module__,
+        )
+        _LOGGER.warning(
+            (
+                "%s::%s implements the `forecast` property or sets "
+                "`self._attr_forecast` in a subclass of WeatherEntity, this is "
+                "deprecated and will be unsupported from Home Assistant 2024.3."
+                " Please %s"
+            ),
+            self.platform.platform_name,
+            self.__class__.__name__,
+            report_issue,
+        )
+
+        translation_placeholders = {"platform": self.platform.platform_name}
+        issue_tracker = async_get_issue_tracker(
+            hass,
+            integration_domain=self.platform.platform_name,
+            module=type(self).__module__,
+        )
+        if issue_tracker:
+            translation_placeholders["issue_tracker"] = issue_tracker
+        ir.async_create_issue(
+            self.hass,
+            DOMAIN,
+            f"deprecated_weather_forecast_{self.platform.platform_name}",
+            breaks_in_ha_version="2024.3.0",
+            is_fixable=False,
+            is_persistent=False,
+            issue_domain=self.platform.platform_name,
+            severity=ir.IssueSeverity.WARNING,
+            translation_key="deprecated_weather_forecast_url"
+            if issue_tracker
+            else "deprecated_weather_forecast_no_url",
```

Comment:
```suggestion
            translation_key=translation_key,
```

We discourage contributors from using the ternary operator, so we should avoid it.

Reply:
ternary operator is best operator, but OK 🤷‍♂️